src/adapters/ui/iconview.js

```diff
@@ -104,17 +105,30 @@ const view = (fileIcon, themeIcon, droppable) => (state, actions) =>
         ]),
         h('div', {
           class: 'osjs-desktop-iconview__entry__label'
-        }, entry.filename)
+        }, (entry.label || entry.filename))
       ])
     ]);
   }));
 
-const createShortcuts = (root, readfile, writefile) => {
+const getApplicationLocaleName = (filename, core) => {
+  const [metadata] = core.make('osjs/packages').getPackages(pkg => (pkg.name === filename));
+  return core.make('osjs/locale').translatableFlat(metadata.title);
+};
+
+const updateShortcutLabel = (shortcut, core) => {
+  shortcut.label = shortcut.label ||
+    (shortcut.mime === 'osjs/application' ? getApplicationLocaleName(shortcut.filename, core) : null);
+
+  return shortcut;
+};
+
+const createShortcuts = (root, readfile, writefile, core) => {
   const read = () => {
     const filename = pathJoin(root, '.shortcuts.json');
 
     return readfile(filename)
       .then(contents => JSON.parse(contents))
+      .then(contents => contents.map(shortcut => updateShortcutLabel(shortcut, core)))
```

Comment:
You can also use a higher order function here:

```
contents.map(updateShortcutLabel(core)))
```

and then in the shortcut updater:

```
const updateShortcutLabel = (core) => (shortcut) => {}
```